scitasexternal/packages/crystal17/package.py

```diff
@@ -0,0 +1,61 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+import os
+
+
+class Crystal17(Package):
+    """Crystal is a code for solid state chemistry and physics."""
+
+    homepage = "http://www.crystal.unito.it/"
+    url      = "file://%s/crystal17-1.0.2.tar.bz2" % os.getcwd()
+
+    # This is not the original file coming from Crystal. This file contains
+    # part of utils17.zip which are scripts used to run Crystal.
+    version('1.0.2', '0ee57f63d400552c46c2dbf1092670d69458c98e6399fc29841ae84cd25c77ac')
+
+    patch('Xmakes-Linux-intel-inc-1.0.2.patch')
+
+    # According to:
+    # http://www.theochem.unito.it/crystal_tuto/mssc2013_cd/tutorials/tuto_HPC/tuto_hpc.html
+    # Pcrystal only scales to 10s of cores and for large systems
+    # one should use MPPcrystal
+    variant('pcrystal', default=True,
+            description='Builds replicated data binary (Pcrystal)')
+    variant('mppcrystal', default=True,
+            description='Builds distributed data binary (MPPcrystal)')
+
+    depends_on('blas')
+    depends_on('lapack')
+    depends_on('mpi', when='+pcrystal')
+    depends_on('mpi', when='+mppcrystal')
+    # Currently only builds with Intel
+    conflicts('%gcc')
```

Comment:
This is not perfect (not specific enough), but it that there is no better solution in spack at the moment to represent "only builds with X".